openwisp_controller/config/base/device.py

```diff
@@ -91,6 +91,13 @@ class AbstractDevice(OrgMixin, BaseModel):
         help_text=_('ip address of the management interface, if available'),
     )
     hardware_id = models.CharField(**(app_settings.HARDWARE_ID_OPTIONS))
+    use_custom_ip = models.BooleanField(
+        default=False,
+        help_text=_(
+            'if enabled, user shall be able to change the '
+            'management ip of the device manually'
+        ),
+    )
```

Comment:
we should not add a new DB column for this.
I think in OpenWISP Monitoring we should remove the field from being readonly, make it readonly with JS unless there's an SNMP check added, in that case we allow to change it.

Reply:
@nemesisdesign but we need some way to track that the user wants to change it, otherwise it will automatically update from the API requests (https://github.com/openwisp/openwisp-controller/pull/496/files#diff-47782e8d264d4907d7893cff2e77b42523d4fb90af8aaef26e78c7a34e996713R75). What do you think?

Reply:
should not be an issue.. openwrt devices will send the right management IP so we can keep it, non openwrt devices will not send management IP and will be empty.

Please verify the following:

- modify your openwrt device to not have "management_interface" in /etc/config/openwisp
- restart openwisp config
- try setting a custom management IP
- restart openwisp config again
- check whether the management IP is modified or not

Reply:
@nemesisdesign in that case, the management IP resets to an empty string.

Reply:
I confirm that the management IP changes after config restarts. I believe this is right. We should add a hint in the admin UI. 

"IP address of the management interface configured on the device. The information received from the device will overwrite any changes done here"

Reply:
Ok thank you @purhan and @pandafy for double checking.

Here's my reasoning.

### 1. OpenWrt devices running openwisp-config and snmpd

In this case, we can assume the management IP will be correctly set if `management_ip` is specified correctly in `/etc/config/openwisp`.

Hence we do not to add code that excludes the management IP update for this case.

### 2. AirOS devices running only snmp deamon, or OpenWrt running only SNMPd

Management IP will not be set at all, nor reset automatically because openwisp-config is not running.

Hence we do not to add code that excludes the management IP update for this case as well.

### This code is not useful

For the reasons explained above I believe this code is not useful.

### What should we look for? What would be useful?

I think we should strive for a solution which is general enough so that it can be flexibly used in different cases.

I believe a solution which can work reliably in many cases is the following:

- allow the field to be writable
- use JS to make it readonly by default, but provide two buttons: one to clear the management IP and one to edit it
  - clear action just resets it to empty string
  - edit action will allow the user to changethe value
- make the help text more detailed, something like @pandafy indicated, eg:

This is the IP address used by OpenWISP to reach the device when performing any type of push operation or active check. The value of this field is generally sent by the device and hence does not need to be changed, but can be changed or cleared manually if needed.

Reply:
@nemesisdesign @pandafy Do you think perhaps a `clear ip` button is redundant? If the user wants to reset it, they can just edit it to be empty? This is how I have made it for now but I can change if you say so.

Reply:
Will check and let you know.

Reply:
> @nemesisdesign @pandafy Do you think perhaps a `clear ip` button is redundant? If the user wants to reset it, they can just edit it to be empty? This is how I have made it for now but I can change if you say so.

It's fine as you did, thanks.